src/iidm/ThreeWindingsTransformerLeg.cpp

```diff
@@ -0,0 +1,215 @@
+
+    const Leg& leg2 = m_transformer.get().getLeg2();
+    count += (leg2.hasPhaseTapChanger() && leg2.getPhaseTapChanger().isRegulating()) ? 1 : 0;
+    count += (leg2.hasRatioTapChanger() && leg2.getRatioTapChanger().isRegulating()) ? 1 : 0;
+
+    const Leg& leg3 = m_transformer.get().getLeg3();
+    count += (leg3.hasPhaseTapChanger() && leg3.getPhaseTapChanger().isRegulating()) ? 1 : 0;
+    count += (leg3.hasRatioTapChanger() && leg3.getRatioTapChanger().isRegulating()) ? 1 : 0;
+
+    return count;
+}
+
+const Terminal& Leg::getTerminal() const {
+    return m_transformer.get().Connectable::getTerminal(m_legNumber - 1);
+}
+
+Terminal& Leg::getTerminal() {
+    return m_transformer.get().Connectable::getTerminal(m_legNumber - 1);
+}
+
+const std::string& Leg::getTypeDescription() const {
+    static std::array<std::string, 3> s_typeDescriptions {{
+                                                              u8"3 windings transformer leg1",
+                                                              u8"3 windings transformer leg2",
+                                                              u8"3 windings transformer leg3"
+                                                          }};
+
+    return s_typeDescriptions[m_legNumber - 1];
+}
+
+double Leg::getX() const {
+    return m_x;
+}
+
+bool Leg::hasPhaseTapChanger() const {
+    return static_cast<bool>(m_phaseTapChanger);
+}
+
+bool Leg::hasRatioTapChanger() const {
+    return static_cast<bool>(m_ratioTapChanger);
+}
+
+CurrentLimitsAdder<const std::nullptr_t, Leg> Leg::newCurrentLimits() {
+    return CurrentLimitsAdder<const std::nullptr_t, Leg>(nullptr, *this);
+}
+
+Leg& Leg::setB(double b) {
+    m_b = checkB(*this, b);
+    return *this;
+}
+
+void Leg::setCurrentLimits(std::nullptr_t /*side*/, std::unique_ptr<CurrentLimits> limits) {
+    m_limits = std::move(limits);
+}
+
+Leg& Leg::setG(double g) {
+    m_g = checkG(*this, g);
+    return *this;
+}
+
+Leg& Leg::setR(double r) {
+    m_r = checkR(*this, r);
+    return *this;
+}
+
+Leg& Leg::setRatedS(double ratedS) {
+    m_ratedS = checkRatedS(*this, ratedS);
+    return *this;
+}
+
+Leg& Leg::setRatedU(double ratedU) {
+    m_ratedU = checkRatedU(*this, ratedU);
+    return *this;
+}
+
+PhaseTapChangerAdder Leg::newPhaseTapChanger() {
+    return PhaseTapChangerAdder(*this);
+}
+
+RatioTapChangerAdder Leg::newRatioTapChanger() {
+    return RatioTapChangerAdder(*this);
+}
+
+Leg& Leg::setX(double x) {
+    m_x = checkX(*this, x);
+    return *this;
+}
+
+void Leg::setPhaseTapChanger(std::unique_ptr<PhaseTapChanger> phaseTapChanger) {
+    m_phaseTapChanger = std::move(phaseTapChanger);
+}
+
+void Leg::setRatioTapChanger(std::unique_ptr<RatioTapChanger> ratioTapChanger) {
+    m_ratioTapChanger = std::move(ratioTapChanger);
+}
+
+void Leg::setTransformer(ThreeWindingsTransformer& transformer) {
+    m_transformer = transformer;
```

Comment:
I think you should add a test to ensure that transformer is not already set like we did for other classes. I think we forgot to do it for TieLines